src/main/java/de/unijena/cheminf/fragment/fingerprint/performanceTest/PerformanceTest.java

```diff
@@ -0,0 +1,419 @@
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * A class for testing the performance of the fragment fingerprinter.
+ *
+ * @author Betuel Sevindik
+ */
+public class PerformanceTest {
+    //<editor-fold defaultstate="collapsed" desc="Private static final constants">
+    /**
+     * Name of file for logging occurred exceptions
+     */
+    private static final String EXCEPTIONS_LOG_FILE_NAME = "Exceptions_Log.txt";
+    /**
+     * Name of file for writing results
+     */
+    private static final String RESULTS_FILE_NAME = "Results.txt";
+    /**
+     * Name of CSV file with the results of the performance test of the bit fingerprints.
+     */
+    private static final String CSV_BIT_FINGERPRINT_PROCESS_RESULT_FILE_NAME = "CSV_BIT_PROCESS_TIME";
+    /**
+     * Name of CSV file with the results of the performance test of the count fingerprints.
+     */
+    private static final String CSV_COUNT_FINGERPRINT_PROCESS_RESULT_FILE_NAME = "CSV_COUNT_PROCESS_TIME";
+    /**
+     * Name of the CSV file with the results of the generated bit arrays.
+     */
+    private static final String BIT_ARRAY_RESULT_FILE_NAME = "BIT_ARRAY_TIME";
+    //</editor-fold>
+    //
+    //<editor-fold defaultstate="collapsed" desc="Private final class variables">
+    /**
+     * Initial capacity of the lists in which the data for generating the fingerprints are stored.
+     */
+    private final int INITIAL_CAPACITY_VALUE = 200000;
+    /**
+     * Separator for separating the lines in the input files
+     */
+    private final String LINE_SEPARATOR_COMMA = ",";
+    /**
+     * Separator for separating the lines in the input files.
+     */
+    private final String LINE_SEPARATOR_SEMICOLON = ";";
+    //</editor-fold>
+    //
+    //<editor-fold defaultstate="collapsed" desc="Private  class variables">
+    /**
+     * The working directory (the jar-file's directory)
+     */
+    private String workingPath;
+    /**
+     * List which is contains all the data of the molecules needed to generate the bit fingerprint.
+     */
+    private ArrayList<ArrayList<String>> moleculeListForBitFingerprint = new ArrayList<>(this.INITIAL_CAPACITY_VALUE);
+    /**
+     * List that contains all fragments, the fingerprint is then generated based on these fragments.
+     */
+    private ArrayList<String> fragmentList = new ArrayList<>(this.INITIAL_CAPACITY_VALUE);
+    /**
+     * Stores the name/ID of the molecules from moleculeFile
+     */
+    private List<String> listOfMoleculeNames;
+    /**
+     * CSV file containing the resulting fragments and their frequencies for each given SMILES (molecule).
+     */
+    private File moleculeFile;
+    /**
+     * CSV file containing the resulting fragments.
```

Comment:
"resulting" indicates that you produce them. Make it clear that this is an input file containing all fragments that are used for initialising the fingerprint.